libraries/AP_IOMCU/AP_IOMCU.cpp

```diff
@@ -724,4 +780,65 @@ void AP_IOMCU::shutdown(void)
     }
 }
 
+/*
+  setup for mixing. This allows fixed wing aircraft to fly in manual
+  mode if the FMU dies
+ */
+bool AP_IOMCU::setup_mixing(RCMapper *rcmap, int8_t override_chan, float mixing_gain)
+{
+    if (config.protocol_version != IOMCU_PROTOCOL_VERSION) {
+        return false;
+    }
+    bool changed = false;
+#define MIX_UPDATE(a,b) do { if ((a) != (b)) { a = b; changed = true; }} while (0)
+
+    // update mixing structure, checking for changes
+    for (uint8_t i=0; i<IOMCU_MAX_CHANNELS; i++) {
+        const SRV_Channel *ch = SRV_Channels::srv_channel(i);
+        if (!ch) {
+            continue;
+        }
+        MIX_UPDATE(mixing.servo_trim[i], ch->get_trim());
+        MIX_UPDATE(mixing.servo_min[i], ch->get_output_min());
+        MIX_UPDATE(mixing.servo_max[i], ch->get_output_max());
+        MIX_UPDATE(mixing.servo_function[i], ch->get_function());
+        MIX_UPDATE(mixing.servo_reversed[i], ch->get_reversed());
+    }
+    // update RCMap
+    MIX_UPDATE(mixing.rc_channel[0], rcmap->roll());
+    MIX_UPDATE(mixing.rc_channel[1], rcmap->pitch());
+    MIX_UPDATE(mixing.rc_channel[2], rcmap->throttle());
+    MIX_UPDATE(mixing.rc_channel[3], rcmap->yaw());
+    for (uint8_t i=0; i<4; i++) {
+        if (mixing.rc_channel[i] <= 0 || mixing.rc_channel[i] >= NUM_RC_CHANNELS) {
```

Comment:
Why do we need to check this here? Can we not rely on `RC_Channels::rc_channel()` to do this same appropriate bounds checking for us and return nullptr?

Reply:
fixed